### PR TITLE
fix(gatsby-source-datocms): pull linked entities on webhook updates

### PR DESCRIPTION
**Pull request by pieh**
This is first attempt at pulling linked entities when Gatsby receive `publish` webhook.

In case of `rich_text` (Modular content) fields as well as `link(s)` we receive just ids of entities to link too, but we don't receive detailed data about those entities so those fields are kept empty. This PR looks to collect linked entity ids and pull them in one go and attach those linked entities with `payload.included` (`entitiesRepo` seems to be prepared to handle those additional entities already)

### CODE REVIEW

File: src/hooks/sourceNodes/index.js [hunk 1]
```diff
@@ -72,6 +72,51 @@ module.exports = async (
...
+                  item_type_rel.id,
+                );
+                const fieldsToResolve = itemTypeForThis.fields.filter(
+                  fieldDef =>
+                    [`rich_text`, `links`, `link`].includes(fieldDef.fieldType),
+                );
+
+                const fieldsToResolveKeys = fieldsToResolve.map(
+                  field => field.apiKey,
+                );
```
Comment (pieh) [inline src/hooks/sourceNodes/index.js:91]: This part looks to get list of fields that response will have only ids and we should pull full entity information

File: src/hooks/sourceNodes/index.js [hunk 2]
```diff
@@ -72,6 +72,51 @@ module.exports = async (
...
+                fieldsToResolveKeys.forEach(f => {
+                  const v = payload.attributes[f];
```
Comment (stefanoverna) [inline src/hooks/sourceNodes/index.js:93]: @pieh this looks fine, but fields can be localized:

```js
const localizedFieldsToResolve = fieldsToResolve.filter(field => f.localized);
```

 In this case `payload.attributes[f]` would be something like this (suppose it's a Links/Rich text field):

```
{
  'en': ['123', '456'],
  'it': ['789'],
}
```

so you would have to additionally iterate over the object keys to get all the IDs for all the locales.

**Review by stefanoverna — CHANGES_REQUESTED** [review]

File: src/hooks/sourceNodes/index.js (hunk 2 above)
Comment (pieh) [reply src/hooks/sourceNodes/index.js:93]: Thanks for this information. I will set up translatable model in my test space to account for this case and will update the PR to cover it

Comment (pieh) [reply src/hooks/sourceNodes/index.js]: Pushed https://github.com/datocms/gatsby-source-datocms/pull/122/commits/e7930ffc5815c0c76b64e0c0f21f812980c24f1a to handle localized fields case (and while I was at it also used some more reasonable variable names than `v` or `acc` in my code ;) )

**Comment by stefanoverna** [comment]
Thanks @pieh, released in 2.4.1!

**Pull request merged**